eth/fetcher/block_fetcher.go

```diff
@@ -417,6 +436,23 @@ func (f *BlockFetcher) loop() {
 			if f.announceChangeHook != nil && len(f.announced[notification.hash]) == 1 {
 				f.announceChangeHook(notification.hash, true)
 			}
+			// if there enable range fetching, just request it and wait for response,
+			// and if it gets timeout and wait for later header & body fetching.
+			if f.enableQuickBlockFetching {
+				go func() {
+					log.Trace("Quick fetching scheduled headers", "peer", notification.origin, "hash", notification.hash)
+					blocks, err := f.fetchRangeBlocks(notification.origin, notification.number, notification.hash, 1)
+					if err != nil {
+						log.Debug("quick block fetching err", "hash", notification.hash, "err", err)
+						return
+					}
+					f.quickBlockFetchingCh <- &BlockFetchingEntry{
```

Comment:
a p2p node may connect thousands of peers.
if parts of them notify a block hash,
then quickly, there will be handreds of goroutine are created
and a lot of copy of the same block are requested.

but quickBlockFetchingCh is consumed one by one, so resulted in goroutine bomb?